src/main/java/burger/list/List.java

```diff
@@ -0,0 +1,223 @@
+package burger.list;
+
+import java.util.ArrayList;
+
+import burger.BurgerException;
+import burger.task.Deadline;
+import burger.task.Event;
+import burger.task.Task;
+import burger.task.Todo;
+
+import static burger.Burger.printLine;
+import static burger.BurgerException.*;
+
+public class List {
+
+    private final ArrayList<Task> taskList;
+
+    public int totalTasks;
+    public List() {
+        taskList = new ArrayList<>();
+        totalTasks = 0;
+    }
+    /**
+     * Returns boolean value base on whether the input matches one of the commands.
+     *
+     *
+     * @param textArray user input in array type.
+     * @return true if it's a valid command and false otherwise.
+     * @throws ArrayIndexOutOfBoundsException when the description after the command is empty.
+     */
+    public boolean isValidCommand(String[] textArray) {
+        boolean isValid = true;
+        try {
+            String command = textArray[0];
+            switch (command) {
+            case "mark":
+                // fallthrough
+            case "unmark":
+                int idx = Integer.parseInt(textArray[1]) - 1;
+                getMark(idx, command);
+                break;
+            case "todo":
+                addTodo(textArray);
+                break;
+            case "deadline":
+                addDeadline(textArray);
+                break;
+            case "event":
```

Comment:
avoid the use of magic literals by storing them in static final vars
https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers